### PR TITLE
Enable Job queueing for CANFAR Science Platform

### PR DESCRIPTION
**Pull request by anujachaitanya**
We worked on [https://jira.skatelescope.org/browse/SP-4266](Enable Job queueing for CANFAR Science Platform
)

- Created helm chart for kueue 
- Modified skaha launch for jobs accordingly 
- Enable feature toggle for GPU resources and limits

@abhishekghoshhh @rob-tw

### CODE REVIEW

File: deployment/helm/skaha/sample-local-values.yaml [hunk 1]
```diff
@@ -2,14 +2,19 @@
 
 # Skaha web service deployment
 deployment:
-  hostname: example.org # change this
+  hostname: example.org # Change this
   skaha:
+    image: images.opencadc.org/platform/skaha:0.16.1
```
Comment (at88mph) [inline deployment/helm/skaha/sample-local-values.yaml]: Please omit this as its already set in the main `values.yaml` file.

File: deployment/helm/skaha/sample-local-values.yaml [hunk 2]
```diff
@@ -2,14 +2,19 @@
...
     imagePullPolicy: IfNotPresent
     # Space delimited list of allowed Image Registry hosts.  These hosts should match the hosts in the User Session images.
     registryHosts: "images.canfar.net"
+    maxUserSessions: "5"
```
Comment (at88mph) [inline deployment/helm/skaha/sample-local-values.yaml]: Please remove this as it's documented in the main `values.yaml` file.

File: deployment/helm/skaha/sample-local-values.yaml [hunk 3]
```diff
@@ -19,7 +24,7 @@ deployment:
 
     # The Resource ID of the Service that contains the Posix Mapping information
     posixMapperResourceID: "ivo://opencadc.org/posix-mapper"
-    # posixMapperResourceID: "ivo://cadc.nrc.ca/gms"
+    registryURL: http://reg.skaha-system.svc.cluster.local/reg
```
Comment (at88mph) [inline deployment/helm/skaha/sample-local-values.yaml]: To enforce users to set this themselves, this should be omitted.

File: deployment/helm/skaha/sample-local-values.yaml [hunk 4]
```diff
@@ -44,19 +49,19 @@ deployment:
 
     # Optionally mount a custom CA certificate
     extraVolumeMounts:
-    # - mountPath: "/config/cacerts"
-    #   name: cacert-volume
+     - mountPath: "/config/cacerts"
+       name: cacert-volume
```
Comment (at88mph) [inline deployment/helm/skaha/sample-local-values.yaml]: Please omit this.  It's optional for users.

File: deployment/helm/skaha/sample-local-values.yaml [hunk 5]
```diff
@@ -44,19 +49,19 @@ deployment:
...
     # Create the CA certificate volume to be mounted in extraVolumeMounts
     extraVolumes:
-    # - name: cacert-volume
-    #   secret:
-    #     defaultMode: 420
-    #     secretName: skaha-cacert-secret
+     - name: cacert-volume
+       secret:
+         defaultMode: 420
+         secretName: science-portal-cacert-secret
```
Comment (at88mph) [inline deployment/helm/skaha/sample-local-values.yaml]: Please omit this as it's optional.

File: deployment/helm/skaha/sample-local-values.yaml [hunk 6]
```diff
@@ -91,3 +96,49 @@ storage:
       # storageClassName: ""
       labels:
         storage: local-storage  # Match this label up with whatever was installed in the base install, or the desired PVC
+
+# Kueue setups
+kueue:
```
Comment (at88mph) [inline deployment/helm/skaha/sample-local-values.yaml:98]: Do the values here represent good defaults?

File: deployment/helm/skaha/templates/skaha-tomcat-deployment.yaml [hunk 7]
```diff
@@ -35,7 +35,9 @@ spec:
         - name: skaha.hostname
           value: "{{ .Values.deployment.hostname }}"
         - name: SKAHA_TLD
-          value: "{{ .Values.deployment.skaha.skahaTld }}"  
+          value: "{{ .Values.deployment.skaha.skahaTld }}"
+        - name: GPU_ENABLED
+          value: "{{.Values.deployment.skaha.gpuEnabled}}"
```
Comment (at88mph) [inline deployment/helm/skaha/templates/skaha-tomcat-deployment.yaml]: Please pad with spaces for consistency.

**Review by at88mph — CHANGES_REQUESTED** [review]

File: deployment/helm/skaha/sample-local-values.yaml (hunk 1 above)
Comment (anujachaitanya) [reply deployment/helm/skaha/sample-local-values.yaml]: Done

File: deployment/helm/skaha/sample-local-values.yaml (hunk 2 above)
Comment (anujachaitanya) [reply deployment/helm/skaha/sample-local-values.yaml]: Removed

File: deployment/helm/skaha/sample-local-values.yaml (hunk 3 above)
Comment (anujachaitanya) [reply deployment/helm/skaha/sample-local-values.yaml]: Done

File: deployment/helm/skaha/sample-local-values.yaml (hunk 5 above)
Comment (anujachaitanya) [reply deployment/helm/skaha/sample-local-values.yaml]: done

File: deployment/helm/skaha/sample-local-values.yaml (hunk 6 above)
Comment (anujachaitanya) [reply deployment/helm/skaha/sample-local-values.yaml:98]: Yes these are good defaults, however resource quotas can be changed according to availability of the resource in the cluster.

File: deployment/helm/skaha/templates/skaha-tomcat-deployment.yaml (hunk 7 above)
Comment (anujachaitanya) [reply deployment/helm/skaha/templates/skaha-tomcat-deployment.yaml]: Done.

File: deployment/helm/skaha/sample-local-values.yaml (hunk 4 above)
Comment (anujachaitanya) [reply deployment/helm/skaha/sample-local-values.yaml]: Done.

**Pull request closed without merging**